File: seed/pydantic-v2/trace/resources/v_2/resources/v_3/resources/problem/types/non_void_function_definition.py
from pydantic import BaseModel
from resources.v_2.resources.v_3.resources.problem.types.non_void_function_signature import NonVoidFunctionSignature
from resources.v_2.resources.v_3.resources.problem.types.function_implementation_for_multiple_languages import FunctionImplementationForMultipleLanguages
from dt import datetime
from core.datetime_utils import serialize_datetime
class NonVoidFunctionDefinition(BaseModel):
    signature: NonVoidFunctionSignature
    code: FunctionImplementationForMultipleLanguages
    class Config:
        frozen = True
        smart_union = True
        json_encoders = {datetime: serialize_datetime}

File: seed/ruby-sdk-v2/trace/lib/non_void_function_definition.rb
# frozen_string_literal: true

module V2
    module Types
        class NonVoidFunctionDefinition < Internal::Types::Model
            field :signature, NonVoidFunctionSignature, optional: true, nullable: true
            field :code, FunctionImplementationForMultipleLanguages, optional: true, nullable: true
        end
    end
end
